internal/activities/add_premis_agent_test.go

```diff
@@ -0,0 +1,129 @@
+package activities_test
+
+import (
+	"os"
+	"testing"
+
+	temporalsdk_activity "go.temporal.io/sdk/activity"
+	temporalsdk_testsuite "go.temporal.io/sdk/testsuite"
+	"gotest.tools/v3/assert"
+	"gotest.tools/v3/fs"
+
+	"github.com/artefactual-sdps/preprocessing-demo/internal/activities"
+	"github.com/artefactual-sdps/preprocessing-demo/internal/premis"
+)
+
+const expectedPREMISWithAgent = `<?xml version="1.0" encoding="UTF-8"?>
+<premis:premis xmlns:premis="http://www.loc.gov/premis/v3" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.loc.gov/premis/v3 https://www.loc.gov/standards/premis/premis.xsd" version="3.0">
+  <premis:agent>
+    <premis:agentIdentifier>
+      <premis:agentIdentifierType valueURI="http://id.loc.gov/vocabulary/identifiers/local">url</premis:agentIdentifierType>
+      <premis:agentIdentifierValue>https://github.com/artefactual-sdps/preprocessing-sfa</premis:agentIdentifierValue>
+    </premis:agentIdentifier>
+    <premis:agentName>Enduro</premis:agentName>
+    <premis:agentType>software</premis:agentType>
+  </premis:agent>
+</premis:premis>
+`
+
+func TestAddPREMISAgent(t *testing.T) {
+	t.Parallel()
+
+	// Transfer with one file (for execution expected to work).
+	transferOneFile := fs.NewDir(t, "",
```

Comment:
Because each of these test directories and path vars are only used once, I think it's more concise and easier to read to put the test definition right into the test structs, e.g.:

```go
{
	name: "Add PREMIS agent for normal content",
	params: activities.AddPREMISAgentParams{
		PREMISFilePath: fs.NewDir(t, "",
			fs.WithFile("something.txt", "1234567899"),
			fs.WithDir("metadata"),
		).Join("metadata", "premis.xml"),
		Agent:          premis.AgentDefault(),
	},
	result:     activities.AddPREMISAgentResult{},
	wantPREMIS: expectedPREMISWithAgent,
},
```